rkt/image/namefetcher.go

```diff
@@ -183,31 +183,39 @@ func (f *nameFetcher) fetchVerifiedURL(app *discovery.App, u *url.URL, a *asc) (
 	o := f.httpOps()
 	ascFile, retry, err := o.DownloadSignature(a)
 	if err != nil {
-		return nil, nil, err
+		return NopReadSeekCloser(nil), nil, err
 	}
-	defer func() { maybeClose(ascFile) }()
+	defer func() {
+		if ascFile != nil {
+			ascFile.Close()
+		}
+	}()
 
 	if !retry {
 		if err := f.checkIdentity(appName, ascFile); err != nil {
-			return nil, nil, err
+			return NopReadSeekCloser(nil), nil, err
 		}
 	}
 
 	aciFile, cd, err := o.DownloadImage(u)
 	if err != nil {
-		return nil, nil, err
+		return NopReadSeekCloser(nil), nil, err
 	}
-	defer func() { maybeClose(aciFile) }()
+	defer func() {
+		if aciFile != nil {
+			aciFile.Close()
+		}
+	}()
```

Comment:
We are returning the `aciFile` here, hence it is the caller's responsibility to close it, I'd remove these lines alltogether.